docs/2024/puzzles/day11.md

```diff
@@ -2,10 +2,244 @@ import Solver from "../../../../../website/src/components/Solver.js"
+- `z)` merge the digits back to a single `Long`, with the inverse of the function used to split them.
+
+After defining the `blink` function, you can now run the simulation for the required steps, in this case 25:
+
+```scala
+def part1(input: String): Int =
+  val stones0 = parse(input)
+  val stones1 =
+    Iterator
+      .iterate(stones0)(blink)
+      .drop(25)
+      .next
+  stones1.size
+```
+
+In the code above, use the `Iterator.iterate(s)(f)` method to repeatadly apply a function `f` to some state `s`, returning the next state. In this case use the parsed stones for state, and the `blink` function.
+
+Then to get the state after 25 iterations, `drop` 25 elements (which includes the initial state, i.e. before any blinks), and take the `next` value. This returns the stone sequence after 25 blinks. Then call `size` to get the number of stones (i.e. the length of the sequence).
+
+### Part 2
+
+All part 2 asked to do is to run the same simulation, but instead for 75 steps. This does not work, likely intentionally, as the required space to represent the sequence grows exponentially, surpassing much more than any conventional home computer.
+
+So, somehow you need to compress the space required to represent the sequence. There are a couple of options:
+- a) bin-packing: represent each element by 2 numbers `[a1,n1,b1,n2,...`, i.e. the value, and then the number of times it appears consecutively. This preserves order, but does not work well if duplicates are not close together. (**such as with this problem**)
+- b) frequency map: only store the distinct numbers in the sequence, associated to their total count. This is not a sorted representation. **HOWEVER** you actually do not need to consider order, despite what the problem hints at. The answer only requires the total stones in the sequence (i.e. forgets order), and also each stone is transformed independently when there is a blink, so there is no interaction between earlier or later stones.
+
+In this case, I used a frequency map, represented with the type `Map[Long, Long]`.
+
+so how does `blink` change?
+
+This time, instead of iterating through each stone in the sequence, you should consider all the stones of a certain value at once. This means you can no longer consider each transformation independently.
+
+What you can do instead is for each kind of stone, update the totals that will appear after the blink, and the blink won't be considered done until you process each kind of stone that appeared in the last step.
+
+So to accomplish this, use `foldLeft`, which within a single blink step will let us iterate the state of the stones. You can even use the same stones value as the initial state because it is immutable - i.e. each update will create a new copy with the necessary changes. Here is the result:
+
+```scala
+def blinkUnordered(stones: Map[Long, Long]): Map[Long, Long] =
+  stones.foldLeft(stones): (stones, stone) =>
```

Comment:
A nit: Do we need to shadow the name `stones`?

Reply:
I renamed it to `nextStones`